plenum/server/monitor.py

```diff
@@ -528,26 +556,34 @@ def isMasterAvgReqLatencyTooHigh(self):
         Return whether the average request latency of the master instance is
         greater than the acceptable threshold
         """
-        avgLatM = self.getAvgLatency(self.instances.masterId)
-        avgLatB = self.getAvgLatency(*self.instances.backupIds)
+        avgLatM = self.getLatencies(self.instances.masterId)
+        avgLatB = {}
+        for lat_item in [self.getLatencies(instId) for instId in self.instances.backupIds]:
+            for cid, lat in lat_item.items():
+                avgLatB.setdefault(cid, []).append(lat)
 
         # If latency of the master for any client is greater than that of
         # backups by more than the threshold `Omega`, then a view change
         # needs to happen
-        for cid, lat in avgLatB.items():
+        for cid, latencies in avgLatB.items():
```

Comment:
Do we really need to check it for each client? Maybe average is enough?
When do we clear latency for each client? Is it a long operation to loop over all cids?

Reply:
Will be changed in https://jira.hyperledger.org/browse/INDY-1599